### ISSUE TITLE
"Error: Migration engine exited" - Cannot run `prisma migrate deploy` on Azure App Service deploy using Run from Zip package

### ISSUE THREAD

**Issue by dev-arvilmena** [prisma/prisma#17002]
### Bug description

Running `prisma migrate deploy` on Azure app service platform using Run from zip package deployment approach returns:

```Error: Migration engine exited.```

**CONTEXT**

I've been trying to make my NodeJS app work on Azure App Service using [Run from ZIP package](https://learn.microsoft.com/en-us/azure/app-service/deploy-run-package) deployment.

I initially encountered when running `npx migrate deploy` with message: 

```ENOENT: no such file or directory, open '/home/site/wwwroot/node_modules/.bin/prisma_fmt_build_bg.wasm'```

I was able to fix it by adjusting the command from `npx migrate deploy` to ```node node_modules/prisma/build/index.js migrate deploy```
thanks to this [guide](https://gist.github.com/0gust1/19ff169231f84d70939943eaf781420e) I found online

But the next problem is it is failing and giving this error message instead:

```Error: Migration engine exited.```

I already tried setting my `binaryTargets` to

```binaryTargets = ["native","debian-openssl-1.1.x"]```

or even just

```  binaryTargets = ["debian-openssl-1.1.x"]```

but I am still getting the same message

### How to reproduce

1. Launch an Azure App Service
2. Set NodeJS version to 18
3. Set "`WEBSITE_RUN_FROM_PACKAGE=1`"
4. Deploy as instructed in[ Run your app in Azure App Service directly from a ZIP package](https://learn.microsoft.com/en-us/azure/app-service/deploy-run-package)

### Expected behavior

Migration should proceed

### Prisma information

```prisma

generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native","debian-openssl-1.1.x"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}


### Environment & setup

- Platform: [Azure App Service](https://learn.microsoft.com/en-us/azure/app-service/)
- OS: Debian GNU/Linux 11 (bullseye) (based from their BASH ui)
- Database: PostgreSQL
- Node.js version: node@v18.12.1


### Prisma Version

```
prisma                  : 4.7.1
@prisma/client          : 4.7.1
```

**Comment by Jolg42** [comment]
Thanks for the issue! This looks like a duplicate of https://github.com/prisma/prisma/issues/15881

**Comment by mersib** [comment]
Were you able to fix this?

**Comment by joostme** [comment]
We are also experiencing the same issue when trying to run `prisma migrate deploy` on our Azure App Service.
The deployment currently uses the method of providing a runnable ZIP package as described here: https://learn.microsoft.com/en-us/azure/app-service/deploy-run-package

We were able to fix the bug described in #15881 by preserving symlinks in our ZIP.
The issue with `Error: Migration engine exited.` sadly still remains.

Do we have the option to toggle some verbose/debug logging to get a clearer picture of the underlying issue?

**Comment by mersib** [comment]
For anyone still having this issue, I was able to fix this issue by changing the log statement in the prisma package
![image](https://user-images.githubusercontent.com/24504669/214545000-ed9e1117-c479-47c6-9bde-bbdb8a0f2d2d.png)
From this to
`throw new Error(`Migration engine exited.${,_e}`);` and hence the error logged that prisma wasn't able to find the respective binaries for the target OS.  In my case the deployment environment was AWS Lambda and for that prisma needs rhel binaries. So I used an env variable during npm i as suggested by prisma docs, and prisma generate to explicitly generate binaries for the target environment like
`PRISMA_CLI_BINARY_TARGETS=darwin-arm64,rhel-openssl-1.0.x npm install` and `PRISMA_CLI_BINARY_TARGETS=darwin-arm64,rhel-openssl-1.0.x npx prisma generate` thus generating the binaries for environment. Note that in your case the value for this env might change as your OS might be different. Not super familiar with azure so won't be able to tell you what binaries you need but with this you can solve your problem

**Comment by janpio** [comment]
Thanks for that context @mersib - and this is a great suggestion on something we can change here - outputting the error should be universally useful here.

The the problem you describe: 

`binaryTargets` in the `generator` block only influences the engine that are downloaded and used for Prisma Client (as you are configuring the `generator client`). `PRISMA_CLI_BINARY_TARGETS` is indeed the correct way to configure the CLI.

Reading between the lines, it could then mean that the users here develop (and build their zip package they upload) on a different "binaryTarget" than what they deploy to on Azure App Service. Is that possible @joostme @dev-arvilmena?

**Comment by mersib** [comment]
Yep, that's exactly the case. Since generally with prisma, you would be running migrations on the db locally and then packaging your functions, Prisma "assumes" that you will only need query engine binaries for the deployment target and thus it only generates that for the target and not the migration and introspection engines.

**Comment by joostme** [comment]
Thanks for the hint @mersib ! That was indeed the issue. I also edited the `index.js` to output the error in the Azure App Service. Setting the `PRISMA_CLI_BINARY_TARGETS` to `debian-openssl-1.1.x` worked for the Azure App Service as they run on Debian Buster.

And @janpio you are correct. We build our app on our own CI pipeline which is running a non-debian host. So the engines didn't match. I also added the  `PRISMA_CLI_BINARY_TARGETS=debian-openssl-1.1.x` variable to our CI pipeline to ensure the correct binary is downloaded and packed in the zip.

**Comment by mersib** [comment]
Hey @janpio apologies for getting back late. In my case it was an AWS Lambda environment but doesn't matter as the error was the same irrespective. I do see you did push a commit to the prisma repo and looking at the code, it exactly counters what the error was and hence it would not need changes. Cheers
Still just for reference I do vaguely remember the error saying something like:
`"Couldn't find xyz engine in abc directory. Searched in ".prisma/xyz , @prisma/xyz, prisma/xyz"`

**Comment by janpio** [comment]
This failing on Lambda is actually new information and interesting. We hope to get the advanced error output int 4.10.0 which will be released later today - then we can learn more about what is going on there. For Azure we kinda understand what is happening - but not for Lambda. The error message you describe is also common, but should happen in a completely different context @mersib.
